File: backend/services/predict.py
import pandas as pd
import json
import pickle

model = pickle.load(open('services/model.pkl', 'rb'))

with open('services/input.json', 'r') as json_file:
    data = json.load(json_file)

print(data)

df = pd.DataFrame([data])

predictions = model.predict(df)

with open('services/output.txt', 'w') as file:
    file.write(str(predictions[0]))

File: backend/services/predict.js
const util = require("util");
const { exec } = require("child_process");
const fs = require("fs");

const execPromise = util.promisify(exec);

const predict = async (data) => {
  fs.writeFileSync("services/input.json", JSON.stringify(data));

  try {
    console.log('oi')
    // Execute a Python script to make predictions using a promise
    const { stdout, stderr } = await execPromise("python services/predict.py");
    
    // Read and send back the prediction results
    const predictions = fs.readFileSync("services/output.txt", "utf-8");
    console.log(predictions)
    return predictions;
  } catch (error) {
    console.error(`Error: ${error.message}`);
    throw new Error("Something went wrong");
  }
};

module.exports = {
  predict,
};
